templates/flavors/cluster-autoscaler/kustomization.yaml

```diff
@@ -0,0 +1,29 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
+  - ../default
+  - ../../addons/cluster-autoscaler
+patches:
+  - target:
+      group: cluster.x-k8s.io
+      version: v1beta1
+      kind: Cluster
+    patch: |-
+      apiVersion: cluster.x-k8s.io/v1beta1
+      kind: Cluster
+      metadata:
+        name: ${CLUSTER_NAME}
+        labels:
+          cluster-autoscaler: "true"
+  - target:
+      group: cluster.x-k8s.io
+      version: v1beta1
+      kind: MachineDeployment
+    patch: |-
+      apiVersion: cluster.x-k8s.io/v1beta1
+      kind: MachineDeployment
```

Comment:
I think we might also need a patch to [remove the replicas field for the MachineDeployment](https://github.com/linode/cluster-api-provider-linode/blob/main/templates/flavors/base/machineDeployment.yaml#L8)?: https://cluster-api.sigs.k8s.io/tasks/automated-machine-management/autoscaling.html?highlight=autoscale#autoscaling-with-clusterclass-and-managed-topologies

Reply:
Yeah, I initially thought we need to do this too, but it turned out we don't actually create any `ClusterClass` resources (at least for the base flavor) and we're not using managed topologies when we do.